File: library/src/main/java/com/github/lany192/box/widget/SquaredRoundedImageView.java
package com.github.lany192.box.widget;

import android.content.Context;
import android.util.AttributeSet;

import com.makeramen.roundedimageview.RoundedImageView;

/**
 * 正方形ImageView
 */
public class SquaredRoundedImageView extends RoundedImageView {

    public SquaredRoundedImageView(Context context) {
        super(context);
    }

    public SquaredRoundedImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SquaredRoundedImageView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        setMeasuredDimension(getMeasuredWidth(), getMeasuredWidth());
    }
}

File: library/src/main/java/com/github/lany192/box/widget/SquaredRoundedImageView.kt
package com.github.lany192.box.widget

import android.content.Context
import android.util.AttributeSet
import com.makeramen.roundedimageview.RoundedImageView

/**
 * 正方形ImageView
 */
class SquaredRoundedImageView : RoundedImageView {

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        setMeasuredDimension(measuredWidth, measuredWidth)
    }
}